### DISCUSSION TITLE
How to JSX to string?

### DISCUSSION THREAD

**Discussion by refinedkontrol** [honojs/hono#3278] [Q&A] [answered]
Hello,

I am looking for a way to convert from Hono JSX (`JSX.Element`) to string.

Example use case: I'm using SVG in JSX to generate several variations of a logo (different colors, padding, stroke width). I'd like to be able to write the SVG string to a file.

I am aware of the SSG helper functions (<https://hono.dev/docs/helpers/ssg>), but I'm wondering if there's a more straightforward way.

Thank you for Hono! Best HTML templating system I know of!

**Comment by nakasyou** [comment, marked as answer]
@refinedkontrol
Try
```tsx
const svg = <svg><title>SVG</title></svg>

const text = await svg.toString()
```

**Reply by refinedkontrol** [reply]
This worked! Thanks!

**Comment by la3rence** [comment]
My solution (works in node.js runtime):

```jsx
import { renderToString } from 'hono/jsx/dom/server';
import MyComponent from './MyComponent.jsx';

const dom = <MyComponent props={data}></MyComponent>;
const text = renderToString(dom);
```
